app/vendor/views.py

```diff
@@ -48,6 +50,53 @@ def new_listing():
     return render_template('vendor/new_listing.html', form=form)
 
 
+@vendor.route('/csv-upload', methods=['GET', 'POST'])
+@login_required
+@vendor_required
+def csv_upload():
+    """Create a new item."""
+    form = NewCSVForm()
+    listings = []
+    if form.validate_on_submit():
+        csv_field = form.file_upload
+        buff = csv_field.data.stream
+        csv_data = csv.DictReader(buff, delimiter=',')
+        #for each row in csv, create a listing
+        current_vendor = Vendor.get_vendor_by_user_id(user_id=current_user.id)
+        for row in csv_data:
+            #cheap way to skip weird 'categorical' lines
+            if (row[current_vendor.product_id_col]).strip().isdigit():
+                safe_price = stripPriceHelper(row[current_vendor.price_col])
+                proposed_listing = Listing.add_csv_row_as_listing(csv_row=row, price=safe_price)
+                queried_listing = Listing.get_listing_by_product_id(product_id=row[current_vendor.product_id_col])
+                if queried_listing:
+                    # case: listing exists and price has not changed
+                    if queried_listing.price == float(safe_price):
+                        listings.append(proposed_listing)
+                    # case: listing exists and price has changed
+                    else:
+                        print 'in here', queried_listing
+                        queried_listing.price = float(safe_price)
+                        proposed_listing.price = float(safe_price)
+                        proposed_listing.updated = 0
+                        listings.append(proposed_listing)
+                        db.session.commit()
+                    #case: listing does not yet exist
+                else:
+                    proposed_listing.updated = 1
+                    listings.append(proposed_listing)
+                    Listing.add_listing(new_listing=proposed_listing)
+    return render_template('vendor/new_csv.html', form=form, listings=listings)
+
+#get rid of those pesky dollar signs that mess up parsing
+def stripPriceHelper(price):
+    if not price[0].isdigit():
```

Comment:
Can we make sure that this will work on any combination of digits until the first real number? e.g. if i have something like <SPACE>$3.99 start at 3.99